todo-app/src/TaskForm.js

```diff
@@ -0,0 +1,61 @@
+import React from 'react'
+import TextField from './TextField'
+import LinkButton from './LinkButton'
+import DatePicker from 'react-datepicker';
+import Moment from 'moment';
+import 'react-datepicker/dist/react-datepicker.css';
+import {observer} from 'mobx-react';
+
+@observer
+export default class TaskForm extends React.Component {
+
+  state = {
+    taskText: '',
+    taskDate: Moment()
+  };
+
+  handleAddTaskClick = () => {
+    this.props.tasksStore.addTask(this.state.taskText , this.state.taskDate.format('YYYY-MM-DD'));
+    this.setState({taskText: '' , taskDate: Moment()});
```

Comment:
You do have duplication of state declaration here, although it's only twice. If you had a third or fourth instance, you could start to think about doing something like this:

    const initialState = {
      taskText: '',
      taskDate: Moment()
    };

    @observer
    export default class TaskForm extends React.Component {

      state = { ...initialState }

      handleAddTaskClick = () => {
        this.props.tasksStore.addTask(this.state.taskText , this.state.taskDate.format('YYYY-MM-DD'));
        this.setState({ ...initialState });
      };